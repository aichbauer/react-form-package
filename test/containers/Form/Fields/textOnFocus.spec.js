import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import {
  Form,
  Field,
} from '../../../../src';

configure({ adapter: new Adapter() });

test('Render Form with text input | onFocus', () => {
  const fieldSpy = jest.spyOn(Form.prototype, 'handleOnFocus');
  const event = {
    target: {
      id: 'text',
      value: '',
    },
  };
  const rules = {
    type: 'text',
    min: '3',
    max: '8',
    required: true,
  };

  const myComponent = (
    <Form>
      <div>
        <div>text</div>
        <Field type="text" id="text" min="3" max="8" required />
      </div>
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);
  let state = wrapper.state();

  expect(state.data.text.value).toBe('');
  expect(state.data.text.valid).toBe(false);

  wrapper.find('input').simulate('focus', event, rules);
  wrapper.update();

  state = wrapper.state();

  expect(state.data.text.value).toBe('');
  expect(state.data.text.valid).toBe(false);
  expect(fieldSpy).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();
});
