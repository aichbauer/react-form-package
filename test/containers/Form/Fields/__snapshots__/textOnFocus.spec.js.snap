// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form with text input | onFocus 1`] = `
<form>
  <div>
    <div>
      text
    </div>
    <input
      bindToAlways={false}
      className="rfp-input"
      dynamic={false}
      id="text"
      invalid={false}
      max="8"
      min="3"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="text"
      valid={false}
      value=""
    />
  </div>
</form>
`;
